### ISSUE TITLE
Error during sending a message

### ISSUE THREAD

**Issue by j-cata** [Vonage/vonage-php-sdk-core#360]
When I try to send a message, the code is trying to assign an array to the property $authHandler of the Vonage\Client\APIResource class, but that property is expecting an object that implements the Vonage\Client\Credentials\Handler\HandlerInterface interface

## Expected Behavior
The message should be sent.

## Current Behavior
I got this error : 
In APIResource.php line 352:                                                                               
Cannot assign array to property Vonage\Client\APIResource::$authHandler of   
type ?Vonage\Client\Credentials\Handler\HandlerInterface

## Steps to Reproduce (for bugs)
```php
$basic = new Vonage\Client\Credentials\Basic($key, $token);
$vonage_client = new Vonage\Client($basic);
vonage_client->sms()->send(
    new Vonage\SMS\Message\Message(array(
        'to' => $phoneNumber,
        'from' => 'Test',
        'text' => $message
    ))
);
```

## Your Environment
Symfony : 6.1.5
PHP : 8.1.11
vonage/client : 4.0.0

**Comment by Froxz** [comment]
@j-cata should be fixed in #361 thanks to @dragonmantank

**Comment by SecondeJK** [comment]
Released as 4.0.4, thanks to @dragonmantank for hopping onto this outside of EMEA hours
